### ISSUE TITLE
miscompile of non-canonical add+icmp by arm64 backend

### ISSUE THREAD

**Issue by regehr** [llvm/llvm-project#55627]
```llvm
define i1 @f() {
  %1 = add i8 0, 159
  %2 = icmp ugt i8 243, %1
  ret i1 %2
}
```
this clearly needs to compile to something emitting true (a few other backends I tried do this) but using top of tree this is what we get from the arm64 backend:
```console
regehr@john-home:~$ llc --march=aarch64 foo.ll -o -
	.text
	.file	"foo.ll"
	.globl	f                               // -- Begin function f
	.p2align	2
	.type	f,@function
f:                                      // @f
	.cfi_startproc
// %bb.0:
	mov	w0, wzr
	ret
.Lfunc_end0:
	.size	f, .Lfunc_end0-f
	.cfi_endproc
                                        // -- End function
	.section	".note.GNU-stack","",@progbits
regehr@john-home:~$ 
```
cc @ornata @nunoplopes @ryan-berger @nbushehri @zhengyang92 @aqjune @Hatsunespica

**Comment by topperc** [comment]
Any chance this is fixed by https://reviews.llvm.org/D125654

**Comment by topperc** [comment]
Fixed by 4638766794b0d9e9a9262e59e3a5cccde657b2ef

**Comment by regehr** [comment]
ah you beat me to it, I was just compiling that :)
